### PR TITLE
FS-1322 New session sign-out endpoints & views

### PR DESCRIPTION
**Pull request by harryyo**
### JWT Session Sign-out added

- New sessions/sign-out route added to authenticator api, which clears the jwt cookie, redis magic-link/account record and redis session record, and then redirects to...
- New service/magic-links/signed-out page for redirection back to create a new magic link.
- Models/magic_link.py refactored to facilitate this

### CODE REVIEW

File: api/session/auth_session.py [hunk 1]
```diff
@@ -35,6 +38,58 @@ def user():
...
+                status = "invalid_token"
+
+            if not valid_token:
+                # Check if token has just expired
+                try:
+                    valid_token = decode_with_options(
+                        existing_auth_token, options={"verify_exp": False}
+                    )
+                except jwt.PyJWTError:
+                    status = "expired_token"
```
Comment (srh-sloan) [inline api/session/auth_session.py:70]: Nitpicky - took me a while to understand this so I googled, could you not just catch a `jwt.ExpiredSignatureError` in the first try block and set `expired_token` there? https://pyjwt.readthedocs.io/en/latest/usage.html#expiration-time-claim-exp

**Review by srh-sloan — APPROVED** [review]
All good, one small comment for you :)

File: api/session/auth_session.py (hunk 1 above)
Comment (harryyo) [reply api/session/auth_session.py]: Refactored, thanks Sarah!

File: api/session/auth_session.py [hunk 2]
```diff
@@ -35,6 +39,54 @@ def user():
                 error_response(404, "Session token expired or invalid")
         error_response(404, "No session token found")
 
+    @classmethod
+    def clear_session(cls):
```
Comment (Adam-W1) [inline api/session/auth_session.py:43]: does this need to be a class method?

**Review by gidsg — APPROVED** [review]

File: api/session/auth_session.py [hunk 3]
```diff
@@ -35,6 +39,54 @@ def user():
...
+        # Clear the cookie and redirect to signed out page
+        signed_out_url = url_for("magic_links_bp.signed_out", status=status)
+        response = make_response(redirect(signed_out_url), 302)
+        response.set_cookie(
+            Config.FSD_USER_TOKEN_COOKIE_NAME,
+            "",
+            domain=Config.COOKIE_DOMAIN,
+            expires=0,
+        )
+        return response
```
Comment (Adam-W1) [inline api/session/auth_session.py:88]: nice and clear logic flow ^ 👍

File: api/session/auth_session.py (hunk 2 above)
Comment (harryyo) [reply api/session/auth_session.py]: You're right I don't think it does tbh - refactored to a staticmethod.

**Review by Adam-W1 — APPROVED** [review]

**Pull request merged**